darts/models/forecasting/baselines.py

```diff
@@ -123,6 +123,55 @@ def predict(self, n: int, num_samples: int = 1, verbose: bool = False):
         return self._build_forecast_series(forecast)
 
 
+class NaiveMovingAverage(LocalForecastingModel):
+    def __init__(self, W: int = 1):
+        """Naive Moving Average Model
+
+        This model predicts using predicts using an auto-regressive moving average.
+
+        Parameters
+        ----------
+        W is the size of the sliding window used to calcualte Moving Average
+        """
+        super().__init__()
+        self.W = W
+        self.last_w_vals = None
+
+    @property
+    def min_train_series_length(self):
+        return self.W
+
+    def __str__(self):
+        return f"Naive moving average model, with W={self.W}"
+
+    def fit(self, series: TimeSeries):
+        super().fit(series)
+
+        raise_if_not(
+            len(series) >= self.W,
+            f"The time series requires at least W={self.W} points",
+            logger,
```

Comment:
This test is unnecessary, it will be trigerred in `ForecastingModel.fit()` as you already overrode the `min_train_series_length` property.